### PR TITLE
Form Smart Paste: implement new Form options

### PR DESCRIPTION
**Pull request by pharret31**

### CODE REVIEW

File: packages/devextreme/js/__internal/core/ai_integration/commands/smartPaste.test.ts [hunk 1]
```diff
@@ -102,6 +102,19 @@ describe('SmartPasteCommand', () => {
 
       expect(result).toStrictEqual(expectedResult);
     });
+
+    it('should not include an unexisted fields into parsed result', () => {
```
Comment (anna-shakhova) [inline packages/devextreme/js/__internal/core/ai_integration/commands/smartPaste.test.ts]: Confusing name, as this method knows what fields are valid
suggestion: should not parse empty field data to result or smth like that 

should we also handle case with response
Field1:::value1;;;Field2;;;

File: packages/devextreme/js/__internal/ui/form/form.ts [hunk 2]
```diff
@@ -199,6 +208,9 @@ class Form extends Widget<FormProperties> {
       stylingMode: config().editorStylingMode,
       labelMode: 'outside',
       isDirty: false,
+      aiIntegration: undefined,
```
Comment (anna-shakhova) [inline packages/devextreme/js/__internal/ui/form/form.ts]: discussed in prev PR, let's not set is to undefined, as it will overwrite ..super._getDefaultOptions() result

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.aiIntegration.tests.js [hunk 3]
```diff
@@ -333,4 +333,229 @@ QUnit.module('SmartPaste', () => {
...
+
+            form.itemOption('test', 'aiOptions', { disabled: false, instruction: 'new instruction' });
+            aiOptions = form.itemOption('test').aiOptions;
+
+            assert.deepEqual(aiOptions, { disabled: false, instruction: 'new instruction' });
+        });
+
+        QUnit.test('item with aiOptions.disabled should be excluded from AI processing', function(assert) {
+            const aiIntegration = {
+                smartPaste: (params) => {
```
Comment (anna-shakhova) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.aiIntegration.tests.js]: Duplicates: QUnit.test('Excludes items with aiOptions.disabled=true',...)

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.aiIntegration.tests.js [hunk 4]
```diff
@@ -333,4 +333,229 @@ QUnit.module('SmartPaste', () => {
...
+                    aiOptions: {
+                        disabled: false,
+                    },
+                }],
+            });
+
+            form.smartPaste('text');
+        });
+
+        QUnit.test('instruction should be passed to AI processing', function(assert) {
```
Comment (anna-shakhova) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.aiIntegration.tests.js]: duplicates: QUnit.test('should call smartPaste command with correct field data',

File: packages/devextreme/js/__internal/ui/form/form.ts [hunk 5]
```diff
@@ -204,6 +213,10 @@ class Form extends Widget<FormProperties> {
       stylingMode: config().editorStylingMode,
       labelMode: 'outside',
       isDirty: false,
+      // @ts-expect-error ts-error
+      onSmartPasting: null,
```
Comment (anna-shakhova) [inline packages/devextreme/js/__internal/ui/form/form.ts:217]: please, add tests for default values for new options

**Review by anna-shakhova — DISMISSED** [review]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.aiIntegration.tests.js [hunk 6]
```diff
@@ -622,4 +622,182 @@ QUnit.module('SmartPaste', () => {
             });
         });
     });
+
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.aiIntegration.tests.js:625]: Suggestions for tests:

1. Edge cases for invokeConditionally:

Handling rejected Promise in cancel
Behavior when cancelCallback is undefined

2. Integration tests:

Verify complete flow: smartPaste -> onSmartPasting -> (cancel=false) -> field updates -> onSmartPasted

3. AIResult typing:

Tests for array values (string[]) in aiResult

**Review by EugeniyKiyashko — DISMISSED** [review]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.aiIntegration.tests.js (hunk 6 above)
Comment (pharret31) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.aiIntegration.tests.js:625]: 1. Added test for rejected promise just in case, the second edge case is covered by invokeConditionally tests
2. integration tests seems should be written after we implement handling issues on pasting
3. Added test for arrays in SmartPaste module

**Review by anna-shakhova — APPROVED** [review]
